pkg/component/component.go

```diff
@@ -226,6 +226,9 @@ func (r *RuntimeSpecs) PolicyToComponents(policy map[string]interface{}) ([]Comp
 				// allows individual inputs (like endpoint) to detect policy changes more easily.
 				injectInputPolicyID(policy, input.input)
 
+				// Propagate type into streams so it can be picked up by beats.
+				copyStreamTypeIntoStreams(policy, input.input)
```

Comment:
I think my general problem with this is that the Beats should be dealing with this correctly.

I think we want the configuration sent to Beats to match what was in the agent policy as closely as possible. This was one of the principles of V2.

Reply:
i agree with you, i would also consider handling this here as we have many dependent processes and we need to make sure everybody does the correct thing. if somebody forgets to deal with it it will be us handling initial investigation during customer escalation. 
you see we're running into this with winlog and synthetics already

i'd rather be a bit defensive here

Reply:
Well, we (as in the Beats data plane team) did the original synthetics and winlog changes which is why I am leaning towards fixing this there. The individual Beat teams generally didn't touch any of the configuration logic, we own both sides of this.

Reply:
This should be fixed in the beats, this is not an Elastic Agent side problem. The goal of V2 is that Elastic Agent doesn't need or even want to know the details of how the configuration is used.

We need to stick to that goal!